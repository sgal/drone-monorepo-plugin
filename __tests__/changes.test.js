describe("changes", () => {
    describe("hasChangesetTrigger", () => {
        it("should return false if changeset param is null", () => {
            const changes = require("../changes");
            const result = changes.hasChangesetTrigger(null);

            expect(result).toBe(false);
        });

        it("should return false if changeset param is an empty array", () => {
            const changes = require("../changes");
            const result = changes.hasChangesetTrigger([]);

            expect(result).toBe(false);
        });

        it("should return true if changeset param is non-empty array", () => {
            const changes = require("../changes");
            const result = changes.hasChangesetTrigger([1]);

            expect(result).toBe(true);
        });

        it("should return false if changeset param is an object with no 'include' or 'exclude' keys", () => {
            const changes = require("../changes");
            const result = changes.hasChangesetTrigger({});

            expect(result).toBe(false);
        });

        it("should return false if changeset's 'include' value is empty", () => {
            const changes = require("../changes");
            const result = changes.hasChangesetTrigger({include: []});

            expect(result).toBe(false);
        });

        it("should return false if changeset's 'exclude' value is empty", () => {
            const changes = require("../changes");
            const result = changes.hasChangesetTrigger({exclude: []});

            expect(result).toBe(false);
        });

        it("should return true if changeset's 'include' value is non-empty", () => {
            const changes = require("../changes");
            const result = changes.hasChangesetTrigger({include: [1]});

            expect(result).toBe(true);
        });

        it("should return true if changeset's 'exclude' value is non-empty", () => {
            const changes = require("../changes");
            const result = changes.hasChangesetTrigger({exclude: [1]});

            expect(result).toBe(true);
        });

        it("should ignore 'exclude' value if 'include' is present", () => {
            const changes = require("../changes");
            const result = changes.hasChangesetTrigger({include: [], exclude: [1]});

            expect(result).toBe(false);
        });
    });

    describe("getIncludeTriggers", () => {
        it("should return triggerConfig param if it is an array", () => {
            const changes = require("../changes");
            const trigger = [];
            const result = changes.getIncludeTriggers(trigger);

            expect(result).toBe(trigger);
        });

        it("should return triggerConfig's 'include' value if trigger is an object", () => {
            const changes = require("../changes");
            const trigger = {include: []};
            const result = changes.getIncludeTriggers(trigger);

            expect(result).toBe(trigger.include);
        });
    });
});
