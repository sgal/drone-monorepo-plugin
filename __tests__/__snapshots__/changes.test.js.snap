// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes getMatchingConfig should disable pipeline which changeset trigger does not match changed files 1`] = `
"kind: pipeline
name: build
steps:
  - name: build
    image: node:10.15.3
    commands:
      - export TZ=Europe/Stockholm
      - curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.16.0
      - export PATH=\\"$HOME/.yarn/bin:$PATH\\"
      - yarn install --ignore-optional --frozen-lockfile
      - yarn build
trigger:
  event:
    exclude:
      - \\"*\\"

---
kind: secret
name: access_id
get:
  path: prod/aws
  name: access_key_id
"
`;

exports[`changes getMatchingConfig should filter out steps which changeset trigger does not match the changes 1`] = `
"kind: pipeline
name: build
steps:
  - name: build
    image: node:10.15.3
    commands:
      - npm run build
    when:
      changeset:
        - \\"*.js\\"
trigger:
  branch:
    - master
  event:
    - push
"
`;

exports[`changes getMatchingConfig should replace pipelines where all steps are filtered out with noop config 1`] = `
"kind: pipeline
  name: default_0
  trigger:
    event:
      exclude: [ \\"*\\" ]
"
`;

exports[`changes getMatchingConfig should return config with no changes if event trigger does not match the drone event 1`] = `
"kind: pipeline
name: build
steps:
  - name: build
    image: node:10.15.3
    commands:
      - export TZ=Europe/Stockholm
      - curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.16.0
      - export PATH=\\"$HOME/.yarn/bin:$PATH\\"
      - yarn install --ignore-optional --frozen-lockfile
      - yarn build
trigger:
  branch:
    - master
  event:
    - push

---
kind: secret
name: access_id
get:
  path: prod/aws
  name: access_key_id
"
`;

exports[`changes getMatchingConfig should return config with no changes if no changeset triggers are present 1`] = `
"kind: pipeline
name: build
steps:
  - name: build
    image: node:10.15.3
    commands:
      - export TZ=Europe/Stockholm
      - curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.16.0
      - export PATH=\\"$HOME/.yarn/bin:$PATH\\"
      - yarn install --ignore-optional --frozen-lockfile
      - yarn build
trigger:
  branch:
    - master
  event:
    - push

---
kind: secret
name: access_id
get:
  path: prod/aws
  name: access_key_id
"
`;
